compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -229,7 +229,14 @@ impl<'hir> Map<'hir> {
                 ItemKind::Static(..) => DefKind::Static,
                 ItemKind::Const(..) => DefKind::Const,
                 ItemKind::Fn(..) => DefKind::Fn,
-                ItemKind::Macro(..) => DefKind::Macro(MacroKind::Bang),
+                ItemKind::Macro(..) => DefKind::Macro(
+                    self.tcx
+                        .resolutions(())
+                        .macro_kind_overrides
+                        .get(&local_def_id)
+                        .copied()
+                        .unwrap_or(MacroKind::Bang),
```

Comment:
Yes, I think that would be better.
I assumed that with something like https://github.com/rust-lang/rust/pull/90451 it doesn't matter because the tables used by lowering will still need to live in `ResolverOutputs`, but if there are multiple resolver output structures then there's still a potential to put this table to some more short living structure.